dvc/pkg.py

```diff
@@ -0,0 +1,160 @@
+import os
+import shutil
+
+from funcy import cached_property
+from schema import Optional
+
+from dvc.config import Config
+from dvc.cache import CacheConfig
+from dvc.path_info import PathInfo
+from dvc.exceptions import DvcException
+from dvc.utils.compat import urlparse, TemporaryDirectory
+
+
+class NotInstalledPkgError(DvcException):
+    def __init__(self, name):
+        super(NotInstalledPkgError, self).__init__(
+            "Package '{}' is not installed".format(name)
+        )
+
+
+class Pkg(object):
+    PARAM_NAME = "name"
+    PARAM_URL = "url"
+    PARAM_VERSION = "version"
+
+    SCHEMA = {
+        Optional(PARAM_NAME): str,
+        Optional(PARAM_URL): str,
+        Optional(PARAM_VERSION): str,
+    }
+
+    def __init__(self, pkg_dir, **kwargs):
+        self.pkg_dir = pkg_dir
+        self.url = kwargs.get(self.PARAM_URL)
+
+        name = kwargs.get(self.PARAM_NAME)
+        if name is None:
+            name = os.path.basename(self.url)
+        self.name = name
+
+        self.version = kwargs.get(self.PARAM_VERSION)
+        self.path = os.path.join(pkg_dir, self.name)
+
+    @cached_property
+    def repo(self):
+        from dvc.repo import Repo
+
+        if not self.installed:
+            raise NotInstalledPkgError(self.name)
+
+        return Repo(self.path, version=self.version)
+
+    @property
+    def installed(self):
+        return os.path.exists(self.path)
+
+    def install(self, cache_dir=None):
+        import git
+
+        if self.installed:
+            return
+
+        git.Repo.clone_from(
+            self.url, self.path, depth=1, no_single_branch=True
```

Comment:
qq - does it avoid restoring the workspace as well?

Reply:
@shcheklein Not sure what you mean. Could you elaborate?

Reply:
I mean `--no-checkout` option of the `git clone`

Reply:
Ah, nope. So we do checkout to have a default version installed. If any dependency is using some particular version, it will get it. But if you've installed some package of specific version by default, then all dependencies that don't have version set explicitly, will use that default installed version.